code-challenges.test.js

```diff
@@ -16,39 +16,136 @@
+//             // how to return the remaining letters
+//         else outPut.push(makeArray[i])
+//     }
+//     return outPut.join("")
+
+// }
+// var secretCodeWord1 = "Lackadaisical"
+// console.log(someTimesY(secretCodeWord1))
+// // Expected output: "L4ck4d41s1c4l"
+// var secretCodeWord2 = "Gobbledygook"
+// console.log(someTimesY(secretCodeWord2))
+// // Expected output: "G0bbl3dyg00k"
+// var secretCodeWord3 = "Eccentric"
+// console.log(someTimesY(secretCodeWord3))
+// // Expected output: "3cc3ntr1c"
+
+// const someTimesY = (string)=>{
+//     outPut = []
+//     makeArray = (string).split("")
+//     for (let i = 0; i < makeArray.length; i++) {
+//         if(makeArray[i] == "a" || makeArray[i] == "A")
+//             outPut.push("4")
+//         else if (makeArray[i] == "e"|| makeArray[i] == "E")
+//             outPut.push("3")
+//         else if (makeArray[i] == "i"|| makeArray[i] == "I")
+//             outPut.push("1")
+//         else if (makeArray[i] == "o"|| makeArray[i] == "O")
+//             outPut.push("0")
+//         else (makeArray[i] !== "a" || makeArray[i] !== "A" || makeArray[i] !== "i"|| makeArray[i] !== "I" || makeArray[i] !== "o"|| makeArray[i] !== "O" || makeArray[i] !== "e"|| makeArray[i] !== "E")
+//             outPut.push(makeArray[i])
+//     }
+//     return outPut
+
+// }
 
 // b) Create the function that makes the test pass.
-
+//////////////////////
+//////////////////////
+//////////////////////
+//////////////////////
 
 
 // --------------------2) Create a function that takes in an array and returns all the words that contain the letter a.
-
 // a) Create a test with an expect statement using the variable provided.
-
-var arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-// Expected output: ["Apple", "Banana", "Orange"]
-var arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-// Expected output: ["Mango", "Apricot", "Peach"]
+// Create a shell test with 
+    // test will take in fruitPicker array
+    // test will .map or forloop for all words that contain a in the name.
+//fn will take in array of words and return only workds that have a
+// maybe a filter method?
+
+// describe("only words with a's", () => {
+//     it("returtakes in an array and returns all the words that contain the letter a", () => {
+//       expect(fruitPicker(["Apple", "Banana", "Plum", "Orange", "Kiwi"])).toEqual(["Apple", "Banana", "Orange"])
+//       expect(fruitPicker(["Mango", "Cherry", "Apricot", "Blueberry", "Peach"])).toEqual(["Mango", "Apricot", "Peach"])
+//     })
+// })
+// const fruitPicker = (array) =>{
+//     //.map wont work.  I tried for loop but I'm having a hard time seperating the a's
+//     basket = []
+//     for (let i = 0; i < array.length; i++) {
+//         if (array[i] === "a")
+//             basket.push(array[i])//I tried using .filiter so I can value and index but I am have a difficult time nailing the syntex.
+        
+//     }    
```

Comment:
yeah theres a method called .includes that would be helpful here! other wise you've got all the bones!